config.yaml

```diff
@@ -22,7 +22,9 @@ tft:
           - name: measure_cpu
           - name: measure_power
           - name: validate_offload
-        # Secondary network is required for tests 27, 28 and 29
+        # (Optional) Secondary network is required for tests 27, 28 and 29. Defaults to None if not set.
```

Comment:
this `Defaults to None if not set` is not right.

The callers of `secondary_network_nad_or_default` have a default as `f"{namespace}/ocp-secondary"` (where `namespace` itself can be configured in YAML but defaults to `"default"`).

The callers that directly use ` ConfConnection.secondary_network_nad` will do something specific about `None` (`pod_ip = y["status"]["podIP"]`), but saying that the default is "None" is not meaningful. For this case, the description should say something like: `If configured, "secondary_network_nad" is also used to detect the pod IP"`.

But can we take one step back? Why `get_pod_ip()` even does this? A value in YAML having a default value, is for convenience of the user so that they can type fewer settings in their configuration. But the behavior should be the same, whether they explicitly type a value or get the default. `get_pod_ip()` should not care about whether the value is explicitly configured or just happens to be the default. This overloads the meaning of whether the value is configured, in confusing ways. If this should be configurable, then add a boolean key "get_pod_ips_from_secondary_nad" and check that in `get_pod_ip()`. Or quite possibly, the test should figure this out by itself without making it configurable to the user.

Reply:
same in `Task.get_template_args()` which does
```
task.py-        common.dict_add_optional(
task.py-            extra, "secondary_network_nad", self.ts.connection.secondary_network_nad
task.py-        )
```
instead of `secondary_network_nad_or_default`.

This changes behavior when rendering `manifests/allow-egress-mnp.yaml.j2`.

Reply:
ok, we discussed this. So "secondary_network_nad" will just behave differently, depending on the test. The README should explain how it is, roughly:

- for tests 27-29, they always need such a NAD. Thus, if left unconfigured, they use a default of `f"{namespace}/ocp-secondary"`.
- in other cases, if `secondary_network_nad` property is explicitly configured in YAML, some (all? which?) tests will use that to use the secondary interface.

Note that with `tft[*].test_cases = "*"` we can run multiple test cases for one test configuration. How it makes sense to configure/omit `secondary_network_nad` value while running test cases that treat the value differently, is up to the user. Maybe it will be common to have configurations with `test_cases = 27-29` or `test_cases = 0-26,30-100`.